results/17-LV-GenomicVariantTest.Rmd

```diff
@@ -0,0 +1,146 @@
+print(samps)
+
+```
+
+# Retrieve Variant Data
+
+Let's retrieve the LV data and evaluate any correlations between scores and tumor size or patient age
+```{r,warning=FALSE}
+data.with.var<-mp_res%>%subset(specimenID%in%samps)%>%
+  left_join(all.vars,by='specimenID')
+
+tab<-subset(data.with.var,!tumorType%in%c('Other','High Grade Glioma','Low Grade Glioma'))
+
+top.genes=tab%>%group_by(tumorType)%>%
+  mutate(numSamps=n_distinct(specimenID))%>%
+      group_by(tumorType,Hugo_Symbol)%>%
+    mutate(numMutated=n_distinct(specimenID))%>%
+    ungroup()%>%
+  subset(numMutated>2)%>%
+      subset(numMutated<(numSamps-1))%>%
+  select(tumorType,Hugo_Symbol,numSamps,numMutated)%>%distinct()
+
+gene.count=top.genes%>%group_by(tumorType)%>%mutate(numGenes=n_distinct(Hugo_Symbol))%>%select(tumorType,numGenes)%>%distinct()
+
+DT::datatable(gene.count)
+
+```
+## Test significance of each gene/immune population
+
+Now we can loop through every tumor type and gene
+
+```{r, warning=FALSE}
+red.genes<-c("NF1","SUZ12","CDKN2A","EED")##for testing
+
+vals<-tab%>%#subset(Hugo_Symbol%in%red.genes)%>%
+    mutate(mutated=ifelse(is.na(IMPACT),'WT','Mutated'))%>%
+  select(latent_var,tumorType,value,Hugo_Symbol,specimenID,mutated)%>%
+  distinct()%>%
+  spread(key=Hugo_Symbol,value='mutated',fill='WT')
+
+counts<-vals%>%
+  gather(key=gene,value=status,-c(latent_var,tumorType,value,specimenID))%>% 
+    select(latent_var,tumorType,value,gene,specimenID,status)%>%
+    group_by(latent_var,tumorType,gene)%>%
+    mutate(numVals=n_distinct(status))%>%
+    subset(numVals==2)%>%ungroup()
+
+#so now we have only 
+with.sig<-counts%>%ungroup()%>%subset(gene%in%top.genes$Hugo_Symbol)%>%
+    group_by(latent_var,gene)%>%
+  mutate(pval=t.test(value~status)$p.value)%>%ungroup()%>%
+  group_by(latent_var)%>%
+  mutate(corP=p.adjust(pval))%>%ungroup()%>%
+  select(latent_var,gene,pval,corP)%>%distinct()
+
+sig.vals<-subset(with.sig,corP<0.05)
+
+DT::datatable(sig.vals)
+
+```
+
+Interesting! Some genes actually pass p-value correction. What do they look like? Here let's write the messiest possible code to print.
+
+```{r,warning=FALSE}
+
+for(ct in unique(sig.vals$latent_var)){
+    tplot<-sig.vals[which(sig.vals$latent_var==ct),]
+    if(nrow(tplot)==0)
+      next
+    print(tplot)
+    p<-counts%>%
+    subset(latent_var==ct)%>%
+    subset(gene%in%tplot$gene)%>%
+    ggplot(aes(x=gene,y=value,col=status))+
+    geom_boxplot(outlier.shape=NA)+
+    geom_point(position=position_jitterdodge(),aes(group=status))+
+    theme(axis.text.x = element_text(angle = 90, hjust = 1))+
+    ggtitle(paste(ct,'scores'))
+#    if(method=='cibersort')
+#      p<-p+scale_y_log10()
+    print(p)
+  }
+#}
+
+
+```
+
+
+
+## Breaking down by tumor type
+At first glance it seems that a lot of these are separating out cNFs (i.e. mast cell signaling) from other types. However, I'm getting the same error I get in notebook number 11, so am unsure about how to proceed. 
```

Comment:
what error? 

And do you think the separation out by cNF might be due to WGS vs WES in the other samples?

Reply:
The error is due to incomplete coverage of variants withins pecicic tumor types, I think. A `trycatch` would be my defacto way of handling it.